### PR TITLE
Mat 4058/population matching

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-4058](https://jira.cms.gov/browse/MAT-4058)
(Optional) Related Tickets:

### Summary
Test Case/patient Population inter dependencies for Cohort and Proportion

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/createTestCase/CreateTestCase.tsx [hunk 1]
```diff
@@ -143,6 +146,8 @@ const CreateTestCase = () => {
   const [showValidationErrors, setShowValidationErrors] = useState(false);
   const [measurementPeriodStart, setMeasurementPeriodStart] = useState<Date>();
   const [measurementPeriodEnd, setMeasurementPeriodEnd] = useState<Date>();
+  const [changedId, setChangedId] = useState("");
```
Comment (jkotanchik-SB) [inline src/components/createTestCase/CreateTestCase.tsx]:
```suggestion
  const [changedPopulation, setChangedPopulation] = useState("");
```
I had to deep dive down into the `triggerPopChanges` function to figure out Id === population. Suggest renaming.

File: src/util/PopulationsMap.ts [hunk 2]
```diff
@@ -54,3 +55,89 @@ export function getFhirMeasurePopulationCode(population: string) {
...
+    //denominator
+    if (changedId === "denominator") {
+      if (expectedValue === true) {
+        myMap[MeasurePopulation.INITIAL_POPULATION].expected = true;
+      }
+
+      if (expectedValue === false) {
+        myMap[MeasurePopulation.NUMERATOR].expected = false;
+        myMap[MeasurePopulation.DENOMINATOR_EXCLUSION] !== undefined &&
+          (myMap[MeasurePopulation.DENOMINATOR_EXCLUSION].expected = false);
```
Comment (jkotanchik-SB) [inline src/util/PopulationsMap.ts:85]: Would safe navigation work the same way here?
```suggestion
        myMap[MeasurePopulation.DENOMINATOR_EXCLUSION]?.expected = false;
```

Comment (sb-prateekkeerthi) [reply src/util/PopulationsMap.ts:85]: No adding "?" to check if it present or not doesn't work In this case

File: src/components/createTestCase/CreateTestCase.tsx (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/components/createTestCase/CreateTestCase.tsx]: Sure, I'll change it

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**